pyanaconda/modules/timezone/installation.py

```diff
@@ -31,36 +35,81 @@
 class ConfigureTimezoneTask(Task):
     """Installation task for timezone configuration."""
 
-    def __init__(self, sysroot, timezone):
+    def __init__(self, sysroot, timezone, isUTC):
         """Create a new task.
 
         :param sysroot: a path to the root of the installed system
         :param str timezone: time zone to set
+        :param bool isUTC: whether time is UTC or local
         :return: new instance
         """
         super().__init__()
         self._sysroot = sysroot
-        self.timezone = timezone
+        self._timezone = timezone
+        self._isUTC = isUTC
 
     @property
     def name(self):
         return "Configure time zone"
 
-    def run(self):
-        """Perform the actual work of setting up timezone."""
-        # get the DBus proxies
-        timezone_proxy = TIMEZONE.get_proxy()
+    def _write_timezone_config(self, root):
+        """
+        Write timezone configuration for the system specified by root.
 
-        # write out timezone configuration
-        kickstart_timezone = timezone_proxy.Timezone
+        :param root: path to the root
+        :raise: TimezoneConfigurationError
+        """
+        # we want to create a relative symlink
+        tz_file = "/usr/share/zoneinfo/" + self._timezone
+        rooted_tz_file = os.path.normpath(root + tz_file)
+        relative_path = os.path.normpath("../" + tz_file)
+        link_path = os.path.normpath(root + "/etc/localtime")
+
+        if not os.access(rooted_tz_file, os.R_OK):
+            log.error("Timezone to be linked (%s) doesn't exist", rooted_tz_file)
+        else:
+            try:
+                # os.symlink fails if link_path exists, so try to remove it first
+                os.remove(link_path)
+            except OSError:
+                pass
+
+            try:
+                os.symlink(relative_path, link_path)
+            except OSError as oserr:
+                log.error("Error when symlinking timezone (from %s): %s",
+                          rooted_tz_file, oserr.strerror)
```

Comment:
I would abstract to a method also this part. It's standalone code to create symlink. It will be much more readable.

Reply:
After squinting at it hard enough, I think there are actually 3 things done, so it is possible to abstract it all yet better: run = fix data + make symlink + write conf

Reply:
Done in the "refactor" commit.